aria/modeling/utils.py

```diff
@@ -51,74 +53,95 @@ def service_template(self):
         return self.container.service_template
 
 
-def create_inputs(inputs, template_inputs):
+def merge_parameter_values(parameter_values, declared_parameters, forbidden_names=None):
```

Comment:
The original code was superfluous and complex, doing unpacking and re-packing of values, and actually had a bug in one area of it. I think this code is much cleaner!